### PR TITLE
Serialize instance properties

### LINKED ISSUE #861: Add support for instance properties
When opening and saving a Glyphs file, the `properties` section gets removed in each instance as illustrated below.

```diff
-properties = (
-{
-key = familyNames;
-values = (
-{
-language = dflt;
-value = "Proxima Nova ExCn";
-}
-);
-},
-{
-key = styleMapFamilyNames;
-values = (
-{
-language = dflt;
-value = "Proxima Nova ExCn Black";
-}
-);
-},
-{
-key = postscriptFontName;
-value = "ProximaNovaExCn-Black";
-},
-{
-key = preferredFamilyNames;
-values = (
-{
-language = dflt;
-value = "Proxima Nova Extra Condensed";
-}
-);
-}
-);
```

Skärmavbild 2023-02-20 kl  11 07 34

### PR DESCRIPTION
**Pull request by IvanUkhov**
Closes #861.

### CODE REVIEW

File: Lib/glyphsLib/classes.py [hunk 1]
```diff
@@ -3153,6 +3153,7 @@ def _serialize_to_plist(self, writer):
         if writer.format_version > 2:
             writer.writeObjectKeyValue(self, "axes", keyName="axesValues")
         writer.writeObjectKeyValue(self, "exports", condition=(not self.exports))
+        writer.writeObjectKeyValue(self, "properties", condition="if_true")
```
Comment (IvanUkhov) [inline Lib/glyphsLib/classes.py:3156]: No working as of yet. Will continue to investigate.

**Comment by IvanUkhov** [comment]
I have confirmed that this preserves instance properties. The error below is arguably unrelated to the cahnge.

**Comment by anthrotype** [comment]
@IvanUkhov codecov flakiness is unrelated (see https://github.com/codecov/codecov-action/issues/837), right now we can only either keep retrying until the upload of the coverage file is successful or just ignore it..

I'll take a look at the PR, thanks

**Comment by anthrotype** [comment]
> I have confirmed that this preserves instance properties.

**Comment by anthrotype** [comment]
> I have confirmed that this preserves instance properties.

nice. If it's not too much trouble, mind adding a small test?

**Comment by IvanUkhov** [comment]
@anthrotype, I have added a test. Please take a look.

File: tests/builder/instances_test.py [hunk 2]
```diff
@@ -147,6 +147,22 @@ def test_glyphs3_instance_filtering():
     assert len(doc.instances) == 2
 
 
+def test_glyphs3_instance_properties(tmpdir):
+    properties = [0, 0, 1, 1, 1, 1]
```
Comment (anthrotype) [inline tests/builder/instances_test.py:151]:
```suggestion
    expected_num_properties = [0, 0, 1, 1, 1, 1]
```

**Review by anthrotype — APPROVED** [review]

File: tests/builder/instances_test.py (hunk 2 above)
Comment (IvanUkhov) [reply tests/builder/instances_test.py]: Updated according to the suggestion.

**Comment by IvanUkhov** [comment]
The new failure is again unrelated. Would you be able to re-run?

**Comment by anthrotype** [comment]
I'll just merge

**Pull request merged**